.github/workflows/docker_image.yml

```diff
@@ -0,0 +1,127 @@
+name: Docker build and Push
+
+# Trigger the workflow on pull request
+on:
+  pull_request:
+    types: [opened, synchronize, closed]
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    
+    # Checkout the repository to the GitHub Actions runner
+    - uses: actions/checkout@v3
+
+
+    # Set up Python
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+
+    - name: Get branch name
+      id: branch_name
+      run: echo "::set-output name=branch::${GITHUB_HEAD_REF}"
```

Comment:
I read somewhere that `set-output` is going to be deprecated (I think Isaw it in one of the workflow runs). Also, please refer to the following url for the replacement command https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/